File: HandloomAPI/Models/Sales/ProcessOutModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace HandloomAPI.Models.Sales
{
    public class ProcessOut_AddModel
    {
        public long ProcessOutID { get; set; }
        public string ProcessOutNo { get; set; }
        public DateTime ProcessOutDate { get; set; }
        public string ProcessOutType { get; set; }
        public long ProcessID { get; set; }
        public long PartyID { get; set; }
        public string IssueType { get; set; }
        public string WorkType { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int? DueDays { get; set; }
        public string Remarks { get; set; }
        public decimal TotalQty { get; set; }
        public List<ProcessOut_Item_AddModel> Items { get; set; }

    }
    public class ProcessOut_Item_AddModel
    {
        public long ID { get; set; }
        public long WOPlanningItemComponentID { get; set; }
        public long ProcessInItemID { get; set; }
        public string WONo { get; set; }
        public string ProcessInNo { get; set; }
        public long SrNo { get; set; }
        public long ParentProductID { get; set; }
        public long ProductID { get; set; }
        public long ComponentID { get; set; }
        public decimal Qty { get; set; }
        public string Remarks { get; set; }
    }

    public class ProcessOut_WorkOrderPlanning_ViewModel
    {
        public long WOPlanningID { get; set; }
        public string WONo { get; set; }
        public DateTime WODate { get; set; }
        public string WOType { get; set; }
        public string PreparedByName { get; set; }
        public string AssignedToName { get; set; }
        public string AuthorizedByName { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public int? DueDays { get; set; }
        public decimal TotalQty { get; set; }
        public decimal TotalPendingQty { get; set; }
    }
    public class ProcessOut_WorkOrderPlanning_Item_ViewModel
    {
        public long WOPlanningItemComponentID { get; set; }
        public long WOPlanningItemID { get; set; }
        public long WOPlanningID { get; set; }
        public string WONo { get; set; }
        public long ProcessInItemID { get; set; }
        public long ProcessInID { get; set; }
        public string ProcessInNo { get; set; }
        public long SrNo { get; set; }
        public long ParentProductID { get; set; }
        public long ProductID { get; set; }
        public string sku { get; set; }
        public string ProductName { get; set; }
        public decimal Qty { get; set; }
        public decimal UsedQty { get; set; }
        public long ComponentID { get; set; }
        public string ComponentName { get; set; }
        public string Remarks { get; set; }
        public string Photo { get; set; }
        public List<WorkOrderPlanning_SalesOrder_Item_Attribute_ViewModel> AttributeValues { get; set; }
    }

    public class ProcessOut_ViewModel
    {
        public long ProcessOutID { get; set; }
        public string ProcessOutNo { get; set; }
        public DateTime ProcessOutDate { get; set; }
        public string ProcessOutType { get; set; }
        public long ProcessID { get; set; }
        public string ProcessName { get; set; }
        public long PartyID { get; set; }
        public string PartyName { get; set; }
        public string IssueType { get; set; }
        public string WorkType { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int? DueDays { get; set; }
        public string Remarks { get; set; }
        public decimal TotalQty { get; set; }
        public long ModifiedBy { get; set; }
        public string ModifiedByName { get; set; }
        public DateTime ModifiedDate { get; set; }
        public bool IsLocked { get; set; }
        public List<ProcessOut_Item_AddModel> Items { get; set; }

    }
    public class ProcessOut_Item_ViewModel
    {
        public long ID { get; set; }
        public long ProcessOutID { get; set; }
        public long WOPlanningItemComponentID { get; set; }
        public long WOPlanningItemID { get; set; }
        public long WOPlanningID { get; set; }
        public string WONo { get; set; }
        public long ProcessInItemID { get; set; }
        public long ProcessInID { get; set; }
        public string ProcessInNo { get; set; }
        public long SrNo { get; set; }
        public long ParentProductID { get; set; }
        public long ProductID { get; set; }
        public string sku { get; set; }
        public string ProductName { get; set; }
        public long ComponentID { get; set; }
        public string ComponentName { get; set; }
        public decimal Qty { get; set; }
        public string Remarks { get; set; }
        public string Photo { get; set; }
    }

    public class ProcessOut_Filter
    {
        public long ProcessOutID { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
        public string ProcessOutNo { get; set; }
    }
    
    
}

File: TradebullsAngular/src/app/Models/Sales/ProcessOutModel.ts
export class ProcessOut_AddModel {
    ProcessOutID?:number;
    ProcessOutNo?:string;
    ProcessOutDate?:Date;
    ProcessOutType?:string;
    ProcessID?:number;
    PartyID?:number;
    IssueType?:string;
    WorkType?:string;
    StartDate?:Date | null;
    EndDate?:Date | null;
    DueDays:number | null = null;
    Remarks?:string;
    TotalQty?:number;
    Items:ProcessOut_Item_AddModel[] = [];
}

export class ProcessOut_Item_AddModel {
    ID?:number;
    WOPlanningItemComponentID: number = 0;
    ProcessInItemID: number = 0;
    WONo: string = "";
    ProcessInNo: string = "";
    SrNo?:number;
    ParentProductID?:number;
    ProductID?:number;
    sku?: string;
    ProductName?: string;
    ComponentID?:number;
    ComponentName?: string;
    Qty?:number;
    Remarks?:string;
    Photo: string = "";
}

export class ProcessOut_WorkOrderPlanning_ViewModel {
    WOPlanningID?:number;
    WONo?:string;
    WODate?:Date;
    WOType?:string;
    PreparedByName?:string;
    AssignedToName?:string;
    AuthorizedByName?:string;
    StartDate?:Date | null;
    EndDate?:Date | null;
    DueDays:number | null = null;
    TotalQty?:number;
    TotalPendingQty?:number;
}

export class ProcessOut_WorkOrderPlanning_Item_ViewModel{
    WOPlanningItemComponentID: number = 0;
    WOPlanningItemID: number = 0;
    WOPlanningID: number = 0;
    WONo: string = "";
    ProcessInItemID: number = 0;
    ProcessInID: number = 0;
    ProcessInNo: string = "";
    SrNo: number = 0;
    ParentProductID: number = 0;
    ProductID: number = 0;
    sku?: string;
    ProductName?: string;
    Qty: number = 0;
    UsedQty: number = 0;
    ComponentID?: number;
    ComponentName?: string;
    Remarks?: string;
    Photo?: string;
}


export class ProcessOut_ViewModel {
    ProcessOutID?:number;
    ProcessOutNo?:string;
    ProcessOutDate?:Date;
    ProcessOutType?:string;
    ProcessID?:number;
    ProcessName?:string;
    PartyID?:number;
    PartyName?:string;
    IssueType?:string;
    WorkType?:string;
    StartDate?:Date | null;
    EndDate?:Date | null;
    DueDays:number | null = null;
    Remarks?:string;
    TotalQty?:number;
    Items:ProcessOut_Item_ViewModel[] = [];
}

export class ProcessOut_Item_ViewModel {
    ID?:number;
    WOPlanningItemComponentID: number = 0;
    ProcessInItemID: number = 0;
    WONo: string = "";
    ProcessInNo: string = "";
    SrNo?:number;
    ParentProductID?:number;
    ProductID?:number;
    sku?: string;
    ProductName?: string;
    ComponentID?:number;
    ComponentName?: string;
    Qty?:number;
    Remarks?:string;
    Photo: string = "";
}
